preflight.inc.php

```diff
@@ -142,14 +142,21 @@
 		$errors++;
 	}
 
-	if(in_array('mod_rewrite', apache_get_modules())){
-		$tests['mod_rewrite']['state']="good";
-		$tests['mod_rewrite']['message']='mod_rewrite detected';
-		$tests['api_test']['state']="fail";
-		$tests['api_test']['message']="Apache does not appear to be rewriting URLs correctly. Check your AllowOverride directive and change to 'AllowOverride All'";
-	}else{
-		$tests['mod_rewrite']['state']="fail";
-		$tests['mod_rewrite']['message']='Apache is missing the <a href="http://httpd.apache.org/docs/current/mod/mod_rewrite.html">mod_rewrite</a> module and it is required for the API to function correctly.  Please install it.';
+    if (function_exists('apache_get_modules')) {
+        if (in_array('mod_rewrite', apache_get_modules())) {
+            $tests['mod_rewrite']['state']="good";
+            $tests['mod_rewrite']['message']='mod_rewrite detected';
+            $tests['api_test']['state']="fail";
+            $tests['api_test']['message']="Apache does not appear to be rewriting URLs correctly. Check your AllowOverride directive and change to 'AllowOverride All'";
+        } else{
+            $tests['mod_rewrite']['state']="fail";
+            $tests['mod_rewrite']['message']='Apache is missing the <a href="http://httpd.apache.org/docs/current/mod/mod_rewrite.html">mod_rewrite</a> module and it is required for the API to function correctly.  Please install it.';
+        }
+    } else {
```

Comment:
Well, Apache is not even necessary, as long as you emulate the rewriting on /api/v1 and /api/test subfolders.
I was able to get the tool working on NginX for example, with a very few lines of code: 

```
    location ~ ^/opendcim/api/v1 {
        rewrite ^(.*) /opendcim/api/v1/index.php last;
    }
    location ~ ^/opendcim/api/test {
        rewrite ^(.*) /opendcim/api/test/index.php last;
    }
```
I can update the doc on this matter if you are ok with this.